src/App.js

```diff
@@ -1,44 +1,42 @@
 // @flow
 
 import * as React from 'react';
-import { View } from 'react-native';
+import { StyleSheet, View } from 'react-native';
 import DAOApi from 'brewskey.js-api';
-import { Provider } from 'mobx-react';
 import { useStrict as mobxUseStrict } from 'mobx';
 import config from './config';
 import NavigationService from './NavigationService';
-import RootStore from './stores/RootStore';
 import AppRouter from './AppRouter';
 import PourButton from './components/PourButton';
 import NFCModal from './components/modals/NFCModal';
+import AuthStore from './stores/AuthStore';
+import AppSettingsStore from './stores/AppSettingsStore';
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+  },
+});
 
 mobxUseStrict(true);
 
 DAOApi.initializeDAOApi({
   endpoint: `${config.HOST}api/v2/`,
 });
 
-const STYLE = { flex: 1 };
-
-export const rootStore: { [key: string]: Object } = new RootStore();
-const stores = {};
-Object.getOwnPropertyNames(rootStore).forEach((storeName: string) => {
-  stores[storeName] = rootStore[storeName];
-});
-
-// todo Pour Button should be deepr in the router tree
-// or hidden until login, it may be better use floactin action button
-// like in twitter.
 class App extends React.Component<{}> {
+  async componentWillMount() {
+    await AuthStore.initialize();
```

Comment:
I don't want to block this diff but I think it would be fine do run the initialize from the store constructor?

Reply:
I thought about it,  but I need particular order for store initalizations, I can't run appsettings.initalize() before AuthStore.Initialize()